### PR TITLE
299 resolution

### PR DESCRIPTION
**Pull request by bhbraswell**
This branch should have been called 229-resolution because it addresses #229 
So that was dumb. The fix itself is only slightly less dumb. But voila.

Recently merged from batch-scheduler FWIW.

Illustration:
```
gips_project merra -p tave -d 2010-100 -s test/onebad.shp --key shaid -v4 --alltouch --outdir withalltouch

gips_project merra -p tave -d 2010-100 -s test/onebad.shp --key shaid -v4 --outdir noalltouch
```

Also:

```
def submit_a_job():
    import time
    from numpy.random import randint
    from gips.datahandler import geokit_api
    
    name = str('test'+str(randint(10000000)))
    print name
    job_id = geokit_api.submit_job(
        name,
        'merra_tave_tave',
        {'key':'shaid',
         'site': '/home/rbraswell/repo/gips/gips/test/onebad.shp'},
        {'dates': '2010-100'})
    print job_id
    status = None
    while status != 'complete':
        status_all = geokit_api.job_status(job_id)
        status = status_all[0]
        print(status_all)
        time.sleep(5)
    print status
    results = geokit_api.stats_request_results({'job': job_id})
    print results
    return job_id
```

### CODE REVIEW

File: gips/tiles.py [hunk 1]
```diff
@@ -93,20 +93,22 @@ def mosaic(self, datadir, res=None, interpolation=0, crop=False,
...
                     else:
-                        mosaic(images, fout, self.spatial.site)
+                        if alltouch is True:
+                            # get the product res, this will indicate that we want "alltouch"
+                            timg = gippy.GeoImage(filenames[0])
+                            affine = images[0].Affine()
+                            product_res = (affine[1], affine[5])
+                        else:
+                            product_res = None
+                        mosaic(images, fout, self.spatial.site, product_res=product_res)
```
Comment (ags-tolson) [inline gips/tiles.py]: Optional tidying up, only if you have time:  Set `product_res = None` before the conditional, then drop the else clause.  The conditional can be shortened to `if alltouch:`.  Can you drop the `timg = ` line, as `timg` seems unused?

File: gips/utils.py [hunk 2]
```diff
@@ -389,15 +389,27 @@ def mosaic(images, outfile, vector):
...
-    cmd = 'gdal_merge.py -o %s -ul_lr %s %s %s' % (outfile, ullr, nodatastr, " ".join(filenames))
-    result = commands.getstatusoutput(cmd)
-    VerboseOut('%s: %s' % (cmd, result), 4)
+    # might reuse the gdal merge command
+    def gdal_merge(x0, y0, x1, y1):
+        ullr = "%f %f %f %f" % (x0, y0, x1, y1)
+        cmd = 'gdal_merge.py -o %s -ul_lr %s %s %s' % (outfile, ullr, nodatastr, " ".join(filenames))
+        result = commands.getstatusoutput(cmd)
+        VerboseOut('%s: %s' % (cmd, result), 4)
+        return result[0] == 0
```
Comment (ags-tolson) [inline gips/utils.py]: more optional tidying, only if now's the right time:  `VerboseOut` is being transitioned to `verbose_out`.  New code should avoid `"%f %s" % (n, s)` in favor of `"{} {}".format(n, s)`.  I like `return result[0] == 0` though, nicely brief and clear.  : ]

File: gips/tiles.py (hunk 1 above)
Comment (bhbraswell) [inline gips/tiles.py]: OK, done. I don't like the PEP8 "if condition:" thing, but I'm just a scientist, so I changed it.

File: gips/utils.py (hunk 2 above)
Comment (bhbraswell) [inline gips/utils.py]: I'll defer to @ircwaves  on this one.

**Comment by ags-tolson** [comment]
:+1: to merge, but my opinion is based on a less-extensive knowledge of that part of the codebase

**Pull request merged**